sdk/python/kfp/containers/__init__.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the speci
+
+_default_image_builder = None
```

Comment:
Can you just make the global variable public and avoid to have set/get methods, which seems not pythonic?

Like:
`image_builder = ContainerBuilder()`

Reply:
Actually, as discussed, don't make this to be external API. just hide it as part of build_image_from_working_dir func signature.

Reply:
>avoid to have set/get methods

I agree, but as you see this particular `get` method creates the `ContainerBuilder` lazily. The reason is that `ContainerBuilder` constructor immediately tries to access GCS and create a new bucket. When this fails, the exception is thrown. That's not the experience we want for anyone just running `import kfp`. There is a fix for that pending for a week that you can review: https://github.com/kubeflow/pipelines/pull/1970

Reply:
sg